pandas/core/frame.py

```diff
@@ -1362,7 +1361,7 @@ def dot(self, other: Series) -> Series:
     def dot(self, other: DataFrame | Index | ArrayLike) -> DataFrame:
         ...
 
-    def dot(self, other: AnyArrayLike | FrameOrSeriesUnion) -> FrameOrSeriesUnion:
+    def dot(self, other: AnyArrayLike | DataFrame | Series) -> DataFrame | Series:
```

Comment:
A Series is also AnyArrayLike
```suggestion
    def dot(self, other: AnyArrayLike | DataFrame) -> DataFrame | Series:
```

Reply:
ooh, we should add a check for this

Reply:
ok. we can merge this and let the check fix this